ratatui/benches/main/constraints.rs

```diff
@@ -0,0 +1,59 @@
+use std::hint::black_box;
+use std::rc::Rc;
+
+use criterion::{Criterion, criterion_group};
+use ratatui::layout::Constraint::{Fill, Length, Max, Min, Percentage, Ratio};
+use ratatui::layout::{Direction, Layout, Rect};
+
+const SPLIT_BY: u16 = 10;
+
+fn generate_layout(constraint: &str, area: Rect) -> Rc<[Rect]> {
+    Layout::default()
+        .direction(Direction::Vertical)
+        .constraints({
+            let mut cs = Vec::new();
+            for _ in 0..SPLIT_BY {
+                match constraint {
+                    "Fill" => {
+                        cs.push(Fill(1));
+                    }
+                    "Length" => {
+                        cs.push(Length(area.width / SPLIT_BY));
+                    }
+                    "Max" => {
+                        cs.push(Max(area.width / SPLIT_BY));
+                    }
+                    "Min" => {
+                        cs.push(Min(area.width / SPLIT_BY));
+                    }
+                    "Percentage" => {
+                        cs.push(Percentage(100 / SPLIT_BY));
+                    }
+                    "Ratio" => {
+                        cs.push(Ratio(1, SPLIT_BY.into()));
+                    }
+                    _ => todo!(),
+                }
+            }
+            cs
+        })
+        .split(area)
+}
+
+fn constraints_render(criterion: &mut Criterion) {
+    for size in [16, 64, 255] {
```

Comment:
nit: I think 256 is fine here as it's used as a u16 
```suggestion
    for size in [16, 64, 256] {
```